k2/csrc/fsa_equivalent.cc

```diff
@@ -220,22 +238,43 @@ bool IsRandEquivalent(const Fsa &a, const float *a_weights, const Fsa &b,
     // find out that we don't need that version, we will remove flag
     // `top_sorted` and add requirements as comments in the header file.
     CHECK(top_sorted);
-    double sum_a =
-        ShortestDistance<Type>(a_compose_path, a_compose_weights.data());
-    double sum_b =
-        ShortestDistance<Type>(b_compose_path, b_compose_weights.data());
-    if (!DoubleApproxEqual(sum_a, sum_b)) return false;
+    double max_loglike_path_a =
```

Comment:
It's the `ShortestDistance<Type>` loglike that I would have thought you would be using here.. at least, for testing the determinization algorithm's equivalence.  Different algorithms may require slightly different tests though.
Mostly we'll be using the kMaxWeight option anyway.

Reply:
why do we need kLogSum here for determinization? as we are trying to know if there is at least one path with beam for symbol sequence `path` (I thought LogSum tells that there is such a path but actually not)

Reply:
I mean, if we are testing equivalence for purposes of testing the correctness of determinization that has been done with log semiring, what we care about here is whether the log-sum versions of the shortest-path is better than the beam.